source/common/http/conn_manager_impl.cc

```diff
@@ -836,6 +847,18 @@ ConnectionManagerImpl::ActiveStream::requestHeadersForTags() const {
   return connection_manager_.config_.tracingConfig()->request_headers_for_tags_;
 }
 
+void ConnectionManagerImpl::ActiveStream::callHighWatermarkCallbacks() {
+  for (DownstreamWatermarkCallbacks* callbacks : watermark_callbacks_) {
+    callbacks->onAboveWriteBufferHighWatermark();
```

Comment:
Wondering out aloud about object lifetimes and callback validity, this all seems safe because filters will be around as long as the connection manager is. Probably where things get tricky is when you cross from connection manager to connection pool and back with callback relationships and lifetimes..